src/Model/Post.php

```diff
@@ -336,24 +394,39 @@ private static function selectViewForUser(string $view, $uid, array $selected =
 	}
 
 	/**
-	 * Select rows from the post view for a given user
+	 * Select rows from the post-user-view view for a given user
+	 *
+	 * @param integer $uid       User ID
+	 * @param array   $selected  Array of selected fields, empty for all
+	 * @param array   $condition Array of fields for condition
+	 * @param array   $params    Array of several parameters
+	 *
+	 * @return boolean|object
+	 * @throws \Exception
+	 */
+	public static function selectForUser($uid, array $selected = [], array $condition = [], $params = [])
+	{
+		return self::selectViewForUser('post-user-view', $uid, $selected, $condition, $params);
+	}
+
+	/**
+	 * Select rows from the post-view view for a given user
 	 *
 	 * @param integer $uid       User ID
 	 * @param array   $selected  Array of selected fields, empty for all
 	 * @param array   $condition Array of fields for condition
 	 * @param array   $params    Array of several parameters
-	 * @param bool    $user_mode true = post-user-view, false = post-view
 	 *
 	 * @return boolean|object
 	 * @throws \Exception
 	 */
-	public static function selectForUser($uid, array $selected = [], array $condition = [], $params = [], bool $user_mode = true)
+	public static function selectPostsForUser($uid, array $selected = [], array $condition = [], $params = [])
```

Comment:
See here: https://github.com/friendica/friendica/blob/2021.06-rc/src/Model/Post.php#L321-L324

This method contains checks if the user had blocked or ignored the author or owner. This is independent from the data source.

Reply:
So, is the difference between `post-view` and `post-user-view` about public posts versus potentially private posts?

Reply:
`post-view` does contain all posts, just like `post-user-view` but doesn't have individual entries for each user. So you can use it without problems for public posts but shouldn't for private posts since you couldn't check if the user is allowed to see the content.

Reply:
Thank you for the elaboration, can we use this distinction in the names of the different methods?

Reply:
I would like to keep the name of `Post::select`, `Post::selectFirst` (and the other ones) as simple as possible since they are used so often.